### DISCUSSION TITLE
Voicemail Playback Link and transcripts

### DISCUSSION THREAD

**Discussion by RMinayaCar** [amazon-connect/amazon-connect-salesforce-scv#95] [Q&A] [answered]
Hello All,

I've setup the VMX 2 solution successfully but I have only one question/concern. When I see the transcript it captures the whole conversation from start to finish, not only the part where the customer leaves the voicemail, same happens with the playback link. Also the playback link sounds really choppy and you can barely understand the what is being said.

This voicemail is triggered during the Customer queue, I noticed when you call this functionality from  another flow, it works as expected.

**Comment by dougjaso** [comment, marked as answer]
The issue you're having here is that you have started streaming previously and enabled both sides on the conversation, likely as a part of the transcription startup process for SCV. The voicemail setup is  designed to handle a single channel, the "from the customer" option in the Start media streaming block. When you have both channels, they are jumbled together, resulting in a garbled file. Additionally, since you previously started streaming, the timestamps included with the contact trace record now start at the beginning of the call, instead of at the beginning of the voicemail

When using SCV, the Start media streaming and trigger functions should almost always be configured in the agent whisper flow. This reduces issues with voicemail, but more importantly it reduces your overall licensed minutes consumption. By starting the transcription process for SCV in your main contact flow, you are now transcribing the customer through your IVR, and while sitting on hold. Since the current implementation has a 15 minute transcription limit, you're using up that limit just transcribing a customer on hold.

**Reply by RMinayaCar** [reply]
So essentially I shouldn't have a start streaming component in the call flow, just in the agent whisper flow? or how exactly should construct the contact flow for this issue to disappear?

**Reply by dougjaso** [reply]
For normal, non voicemail calls, you should start streaming and do the KVS trigger in the agent whisper. Referencing Salesforce's examples, you could use [Sample_SCV_Inbound](https://github.com/service-cloud-voice/examples-from-doc/blob/main/ContactFlows/Sample_SCV_Inbound), which does not initialize streaming in the main flow. Edit that to set the agent whisper flow, and use the [Sample_SCV_Agent_Whisper_with_Transcription](https://github.com/service-cloud-voice/examples-from-doc/blob/main/ContactFlows/Sample_SCV_Agent_Whisper_With_Transcription) agent whisper, which has the initialization there. Then if you decide to have a customer leave a voicemail, you can use the 'VMXCoreFlow'  contact flow, which is installed by the VMX2 solution, to intialize streaming for the voicemail, as designed.

**Reply by RMinayaCar** [reply]
The issue that I'm facing is that all call flows must have the voicemail attached to the customer queue sequence, so they can leave a voicemail while they are waiting for an agent. So for those scenarios I should move the KVS Trigger to the agent whisper flow and the remove the start streaming, so it only lives in the voicemail side

**Reply by dougjaso** [reply]
No, i think you are misunderstanding. 
**Normal call flow:**

1.   Incoming Call
2.  Main contact flow
3.  Queue Flow
4.  Agent accepts call
5.  Agent whisper flow
     1.  Start media streaming
     2.  KVS Trigger
     3.  Play whisper message
     4.  End flow
6.  Connect to agent

**Voicemail call flow:**
1.   Incoming Call
2.  Main contact flow
3.  Queue Flow
     1.  Customer chooses voicemail
     2. Set voicemail flags
     3. Start streaming
     4. Record Voicemail
     5. End call.

**Reply by RMinayaCar** [reply]
Gotcha!!

One other thing is, I'm receiving two voicemail as cases every time a customer receives a voicemail, I think it is due to the fact that I used to have the old voicemail solution, but I did erase everything regarding that solution or maybe it can be something else?

**Reply by RMinayaCar** [reply]
This issue has been resolved, it can be closed whenever
